minimesos/src/main/java/com/containersol/minimesos/MesosCluster.java

```diff
@@ -62,6 +62,60 @@ public MesosCluster(ClusterArchitecture clusterArchitecture) {
 
         this.containers = clusterArchitecture.getClusterContainers().getContainers();
         clusterId = Integer.toUnsignedString(new SecureRandom().nextInt());
+        for (AbstractContainer container : containers) {
+            container.setClusterId(clusterId);
+        }
+    }
+
+    /**
+     * Recreate a MesosCluster object based on an existing cluster ID.
+     *
+     * @param clusterId the cluster ID of the cluster that is already running
+     */
+    public MesosCluster(String clusterId) {
```

Comment:
This _looks_ like something that should have been a factory method instead?

From http://stackoverflow.com/questions/14694982/why-is-using-side-effects-bad-practice-in-javascript-constructors
> Side effects are lies. Your function promises to do one thing, but it also does other hidden things...they are devious and damaging mistruths that often result in strange temporal couplings and order dependencies.